### PR TITLE
ZVertexFilter: Only cut on FD and CD particles

### PR DESCRIPTION
**Pull request by rtysonCLAS12**
There is no z-vertex position from the FT, z vertex cut shouldn't be applied to electrons in the FT.

Fix is to require abs( status )>=2000 ie a particle is in FD or CD. Validator now only plots particles in FD or CD.

### CODE REVIEW

File: src/iguana/algorithms/clas12/ZVertexFilter/Algorithm.cc [hunk 1]
```diff
@@ -38,19 +38,21 @@ namespace iguana::clas12 {
     particleBank.getMutableRowList().filter([this](auto bank, auto row) {
         auto zvertex = bank.getFloat("vz", row);
         auto pid = bank.getInt("pid", row);
-        auto accept  = Filter(zvertex,pid);
-        m_log->Debug("input vz {} pid {} -- accept = {}", zvertex, pid, accept);
+        auto status = bank.getInt("status", row);
```
Comment (c-dilks) [inline src/iguana/algorithms/clas12/ZVertexFilter/Algorithm.cc:41]: `status` is `short` type; `hipo::bank::getShort` actually returns an `int`, so downstream code shouldn't need any changes.
```suggestion
        auto status = bank.getShort("status", row);
```

File: src/iguana/algorithms/clas12/ZVertexFilter/Algorithm.h [hunk 2]
```diff
@@ -26,10 +26,12 @@ namespace iguana::clas12 {
       void Stop() override;
 
       /// @action_function{scalar filter} checks if the Z Vertex is within specified bounds if pid is one for which the filter should be applied to
+      /// cuts applied to particles in FD or CD (ie not in FT)
```
Comment (c-dilks) [inline src/iguana/algorithms/clas12/ZVertexFilter/Algorithm.h:29]: doxygen won't insert line breaks (unless forced), so add a semicolon
```suggestion
      /// @action_function{scalar filter} checks if the Z Vertex is within specified bounds if pid is one for which the filter should be applied to;
      /// cuts applied to particles in FD or CD (ie not in FT)
```

File: src/iguana/algorithms/clas12/ZVertexFilter/Validator.cc [hunk 3]
```diff
@@ -60,9 +60,10 @@ namespace iguana::clas12 {
     for(auto const& row : particle_bank.getRowList()) {
       double vz = particle_bank.getFloat("vz", row);
       int pdg = particle_bank.getInt("pid", row);
+      int status = particle_bank.getInt("status", row);
```
Comment (c-dilks) [inline src/iguana/algorithms/clas12/ZVertexFilter/Validator.cc:63]:
```suggestion
      int status = particle_bank.getShort("status", row);
```

**Comment by rtysonCLAS12** [comment]
Example plots outputted by the validator (cutting only on electrons between -5 and 5 cm):

image

image

**Review by c-dilks — APPROVED** [review]

**Pull request merged**